core/src/main/scala/scalaz/std/String.scala

```diff
@@ -59,19 +58,19 @@ trait StringFunctions {
       case NonFatal(e) => Failure(e.getMessage)
```

Comment:
This is more of an open question: do we want `parse*` methods to catch exceptions like `NullPointerException`?
Personally I'd prefer we remove the `NonFatal(e)` cases, and deal with null input more explicitly (eg. via `scalaz.Maybe#fromNullable`) if needed.